File: grading_students.rb
# https://www.hackerrank.com/challenges/grading/problem

def gradingStudents(grades)
  grades.map do |n|
    # diff between n and the next multiple of 5
    diff = 5 - (n % 5)
    if n < 38
      n
    elsif diff < 3
      n += diff
    elsif diff >= 3
      n
    end
  end
end

File: grading_students.py
# https://www.hackerrank.com/challenges/grading/problem

def gradingStudents(grades):
    diff = lambda x: 5 - (x % 5)
    g = lambda x: x + 5 - (x % 5)
    rounded_grades = []

    for grade in grades:
        if grade < 38:
            rounded_grades.append(grade)
        elif diff(grade) < 3:
            rounded_grades.append(g(grade))
        elif diff(grade) >= 3:
            rounded_grades.append(grade)

    return rounded_grades
